core/modules/system/system.admin.inc

```diff
@@ -1566,41 +1566,174 @@ function system_site_information_settings_submit($form, &$form_state) {
+
+  $interval = $config->get('cron_safe_threshold');
+  $form['automatic']['cron_safe_threshold'] = array(
+    '#type' => 'value',
+    '#value' => $interval,
+  );
+
+  // Set a sensible default for number and interval.
+  $enabled_default = TRUE;
+  $units_default = 60; // Minutes.
+  $interval_number = 5;
+
+  // Determine enabled state based on stored value of interval.
+  if ($interval === -1) {
+    $enabled_default = FALSE;
+  }
+  $form['automatic']['cron_auto_enabled'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Enabled'),
+    '#default_value' => $enabled_default,
+  );
+
+  // Set options as number of seconds for easy math.
+  $unit_options = array(
+    604800 => t('weeks'),
+    86400 => t('days'),
+    3600 => t('hours'),
+    60 => t('minutes'),
+  );
+
+  if ($enabled_default) {
+    // Determine inerval number based on optional stored unit preference.
+    $units_default = $config->get('cron_interval_units');
+    if ($units_default) {
+      $interval_number = $interval / $units_default;
+    }
+    else {
+      // Otherwise, select the uint that leaves a whole number.
+      foreach ($unit_options as $key => $value) {
+        if (($interval % $key) == 0) {
+          $units_default = $key;
+          $interval_number = $interval / $key;
+          break;
+        }
+      }
+    }
+  }
+
+  $form['automatic']['interval'] = array(
+    '#type' => 'container',
+    '#id' => 'automatic-cron-interval',
+    '#states' => array(
+      'visible' => array(
+        '#edit-cron-auto-enabled' => array('checked' => TRUE),
+      ),
+    ),
+  );
+
+  $number_element = array(
+    '#type' => 'number',
+    '#default_value' => $interval_number,
+    '#min' => 0,
+    '#step' => 1,
+    '#size' => 3,
+    '#attributes' => array('aria-label' => t('Cron run time interval')),
+  );
+
+  // Sort the intervals from shortest to longest for UI.
+  ksort($unit_options, SORT_NUMERIC);
+  $unit_element = array(
     '#type' => 'select',
-    '#title' => t('Run cron every'),
-    '#description' => t('More information about setting up scheduled tasks can be found by <a href="@url">reading the cron tutorial</a>.', array('@url' => url('https://backdropcms.org/cron'))),
-    '#default_value' => config('system.core')->get('cron_safe_threshold'),
-    '#options' => array(0 => t('Never')) + backdrop_map_assoc(array(600, 1800, 3600, 10800, 21600, 43200, 86400, 604800), 'format_interval'),
+    '#default_value' => $units_default,
+    '#options' => $unit_options,
+    '#attributes' => array('aria-label' => t('Cron run time units')),
+  );
+
+  $sentence = 'If the last cron run was more than !number !unit ago, run cron on the next page request.';
+  $form['automatic']['interval']['sentence'] = array(
+    '#type' => 'container',
+    '#theme' => 'cron_interval_sentence',
+    '#children' => array(
+      'number' => $number_element,
+      'unit' => $unit_element,
+      'sentence' => $sentence,
+    ),
+  );
```

Comment:
Setting `#children` manually, especially with a non-element like `$sentence` looks likely to cause unexpected problems. Maybe just set the nested elements normally and set `#sentence` as it's own property.

```suggestion
  $form['automatic']['interval']['sentence'] = array(
    '#type' => 'container',
    '#theme' => 'cron_interval_sentence',
    '#sentence' => $sentence,
  );
  $form['automatic']['interval']['sentence']['number'] = $number_element;
  $form['automatic']['interval']['sentence']['unit'] = $unit_element;
```